roles/openshift_storage_glusterfs/tasks/glusterfs_common.yml

```diff
@@ -108,14 +102,19 @@
       name: "glusterfs-{{ glusterfs_name }}"
       files:
       - "{{ mktemp.stdout }}/glusterfs-storageclass.yml"
-  when:
-  - glusterfs_storageclass or glusterfs_s3_deploy
+  when: glusterfs_storageclass
 
 - import_tasks: glusterblock_deploy.yml
   when: glusterfs_block_deploy
 
 - import_tasks: glusterblock_storageclass.yml
   when: glusterfs_block_storageclass
 
+- name: Check if gluster-s3 can't be deployed
```

Comment:
This whole set_fact seems unneeded?  Probably just check the reverse of these conditions on the import_tasks below?

Reply:
Fair enough. Something like:
```
when:
- glusterfs_s3_deploy
- glusterfs_s3_account is defined
- glusterfs_s3_user is defined
- glusterfs_s3_password is not defined
```
Note that `glusterfs_s3_deploy` is initially set [here](https://github.com/openshift/openshift-ansible/blob/a5ae3dd53750b72e084d27ac57be72d902870296/roles/openshift_storage_glusterfs/defaults/main.yml#L42), we were just overriding it if the rest of the config wasn't set.

Reply:
yes, your example seems suitable.  set_fact'ing the same variable in multiple places makes troubleshooting really hard (see glusterfs_cli for example ;)

Reply:
* glusterfs_s3_password is defined

Reply:
Good catch, fixed just in case.

Reply:
@jarrpa @michaelgugino  

removed set_fact and added the checks as mentioned.